nixos/modules/services/networking/nebula.nix

```diff
@@ -200,6 +200,10 @@ in
             after = [ "basic.target" "network.target" ];
             before = [ "sshd.service" ];
             wantedBy = [ "multi-user.target" ];
+            # Restart nebula when the config file changes.
+            # Without this, we may get "Current command vanished from the unit file"
+            # when the file changes and systemd will _not_ restart it.
+            restartTriggers = [ configFile ];
```

Comment:
I am not sure what "Current command vanished from the unit file" means and how it affects service restarting.

However, I do not think adding `restartTriggers` makes any difference.  All `restartTriggers` does is adding `X-Restart-Triggers = your-triggers` to the service file.
- Since this directive starts with `X-`, systemd ignores it.
- Also, the logic of restarting services in `switch-to-configuration.pl` does not do anything special for it.  And `configFile` has already been used in `ExecStart`.  So there is no difference after add `restartTriggers` from the perspective of `switch-to-configuration.pl`.

Reply:
> I am not sure what "Current command vanished from the unit file" means and how it affects service restarting.

It is an error/issue with systemd, see e.g. #49415. Sometimes, when a systemd service's unit file changes (e.g. when the command line changes as part of a NixOS configuration switch), there is a race where systemd notices that change, and then seems to run some "somebody changed my file under me, I'll do nothing about it" logic. Unfortunately it isn't documented in systemd, thus my vague explanation.

The effect of that is that sometimes on NixOS configuration switches, services aren't restarted (#49415), and in this case for Nebula, I noticed that when I change the Nebula config file via `configuration.nix`, Nebula was not restarted (thus not observing the config change).

> * the logic of restarting services in `switch-to-configuration.pl` does not do anything special for it.  And `configFile` has already been used in `ExecStart`.  So there is no difference after adding `restartTriggers` from the perspective of `switch-to-configuration.pl`.

Hm, this is correct. From https://github.com/NixOS/nixpkgs/commit/b606165bd9d4c8a1f82160f67bd541f064ed1ddc the only thing `X-Restart-Triggers` does is to be _present_ in the unit file, so that the unit file changes. So you're right that should already tbe the case due to `ExecStart = "${netCfg.package}/bin/nebula -config ${configFile}";`.

I am wondering now whether I was just lucky regarding the race and this change indeed had no additional effect on my systems, or whether the fact that a string in the unit file changes that's outside the `ExecStart` line ("current _command_ ...") helps systemd avoid the problem somehow.

I'm inclined to close this PR, until we have some proof that the change actually does something.